unison-core/src/Unison/Term.hs

```diff
@@ -956,14 +960,14 @@ let1_ isTop bindings e = foldr f e bindings
 
 -- | annotations are applied to each nested Let expression
 let1 ::
-  Ord v =>
+  (Ord v, Semigroup a) =>
   IsTop ->
   [((a, v), Term2 vt at ap v a)] ->
   Term2 vt at ap v a ->
   Term2 vt at ap v a
 let1 isTop bindings e = foldr f e bindings
   where
-    f ((ann, v), b) body = ABT.tm' ann (Let isTop b (ABT.abs' ann v body))
+    f ((ann, v), b) body = ABT.tm' (ann <> ABT.annotation body) (Let isTop b (ABT.abs' (ABT.annotation body) v body))
```

Comment:
This annotation appears to be just incorrect; the body of the let should be annotated with the annotation for the body, not the lhs of the binding.